### PR TITLE
Implements new indicator - CKS - ChandeKrollStop

### LINKED ISSUE #8121: Implement Chande Kroll Stop
#### Expected Behavior
Supports ChandeKrollStop (see [talipp implementation](https://github.com/nardew/talipp/blob/main/talipp/indicators/ChandeKrollStop.py)).

Refer to [LEAN Engine > Contributions > Indicators](https://www.quantconnect.com/docs/v2/lean-engine/contributions/indicators).

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by femtotrader**
#### Related Issue

Closes #8121

#### Motivation and Context

This indicator wasn't implemented in QuantConnect Lean

#### Requires Documentation Change

Docstrings are written but there is probably room for better documentation

#### How Has This Been Tested?

Comparison with talipp computation of ChandeKrollStop(5, 2.0, 3) of SPY candles using [generate_reference_data_from_talipp.py](https://github.com/QuantConnect/Lean/blob/master/Tests/TestData/generate_reference_data_from_talipp.py)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm/QCAlgorithm.Indicators.cs [hunk 1]
```diff
@@ -445,7 +445,6 @@ public CommodityChannelIndex CCI(Symbol symbol, int period, MovingAverageType mo
             return commodityChannelIndex;
         }
 
-        /// <summary>
```
Comment (Martin-Molinero) [inline Algorithm/QCAlgorithm.Indicators.cs:448]: typo

File: Indicators/ChandeKrollStop.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        /// Initializes a new instance of the <see cref="ChandeKrollStop"/> class.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="atrPeriod">The period over which to compute the average true range.</param>
+        /// <param name="atrMult">The ATR multiplier to be used to compute stops distance.</param>
+        /// <param name="period">The period over which to compute the max of high stop and min of low stop.</param>
+        public ChandeKrollStop(string name, int atrPeriod, decimal atrMult, int period)
+            : base(name)
+        {
+            WarmUpPeriod = 1;
```
Comment (Martin-Molinero) [inline Indicators/ChandeKrollStop.cs:75]: > WarmUpPeriod = 1;

Does not seem right? Strange that tests didn't catch this, there might be some other bug there

File: Indicators/ChandeKrollStop.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+            _high_stop_list = new RollingWindow<decimal>(period);
+            _low_stop_list = new RollingWindow<decimal>(period);
```
Comment (Martin-Molinero) [inline Indicators/ChandeKrollStop.cs:78]: Name style is off

File: Indicators/ChandeKrollStop.cs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        protected override decimal ComputeNextValue(IBaseDataBar input)
+        {
+            _atr.Update(input);
+            _inputValues.Add(input);
+
+            var highs = _inputValues.Select(input => input.High).ToList();
+            var high_stop = highs.Max() - _atr.Current.Value * _atrMult;
+            _high_stop_list.Add(high_stop);
+
+            var lows = _inputValues.Select(input => input.Low).ToList(); ;
```
Comment (Martin-Molinero) [inline Indicators/ChandeKrollStop.cs:102]: Could use Min/Max indicators avoid iterating/tolist on ever data point, same for _low_stop_list/_high_stop_list below

File: Indicators/ChandeKrollStop.cs [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data.Market;
+using System;
```
Comment (Martin-Molinero) [inline Indicators/ChandeKrollStop.cs:17]: not used

File: Indicators/ChandeKrollStop.cs [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+        /// <summary>
+        /// Gets the short stop of ChandeKrollStop.
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> ShortStop { get; }
+
+        /// <summary>
+        /// Gets the long stop of ChandeKrollStop.
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> LongStop { get; }
```
Comment (Martin-Molinero) [inline Indicators/ChandeKrollStop.cs:42]: Maybe we can just rename LongStop->Long & ShortStop->Short

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Hey @femtotrader! Thank you, leaving some requests

File: Indicators/ChandeKrollStop.cs (hunk 6 above)
Comment (femtotrader) [reply Indicators/ChandeKrollStop.cs]: ca1720 / identifier contains type name

so keeping as it is

File: Indicators/ChandeKrollStop.cs (hunk 4 above)
Comment (femtotrader) [reply Indicators/ChandeKrollStop.cs]: Done

File: Indicators/ChandeKrollStop.cs (hunk 5 above)
Comment (femtotrader) [reply Indicators/ChandeKrollStop.cs]: Removed

File: Algorithm/QCAlgorithm.Indicators.cs (hunk 1 above)
Comment (femtotrader) [reply Algorithm/QCAlgorithm.Indicators.cs]: Fixed typo

File: Indicators/ChandeKrollStop.cs (hunk 2 above)
Comment (femtotrader) [reply Indicators/ChandeKrollStop.cs:73]: You are right this shouldn't be 1. With `ChandeKrollStop(5, 2.0m, 3);` we are only having ouputs after 7 incoming data but if I put

File: Indicators/ChandeKrollStop.cs [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        /// <returns>The input is returned unmodified.</returns>
+        protected override decimal ComputeNextValue(IBaseDataBar input)
+        {
+            _atr.Update(input);
+
+            _highStopMaximum.Update(input.EndTime, input.High);
+            var high_stop = _highStopMaximum.Current.Value - _atr.Current.Value * _atrMult;
+
+            _lowStopMinimum.Update(input.EndTime, input.Low);
+            var low_stop = _lowStopMinimum.Current.Value + _atr.Current.Value * _atrMult;
```
Comment (Martin-Molinero) [inline Indicators/ChandeKrollStop.cs:97]: Some minor naming improvements:
`low_stop/high_stop` -> `lowStop/highStop`

`_highStopMaximum/_lowStopMinimum` -> `_underlyingMaximum/_underlyingMinimum`

File: Indicators/ChandeKrollStop.cs [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        /// <param name="name">The name.</param>
+        /// <param name="atrPeriod">The period over which to compute the average true range.</param>
+        /// <param name="atrMult">The ATR multiplier to be used to compute stops distance.</param>
+        /// <param name="period">The period over which to compute the max of high stop and min of low stop.</param>
+        public ChandeKrollStop(string name, int atrPeriod, decimal atrMult, int period)
+            : base(name)
+        {
+            WarmUpPeriod = atrPeriod + period - 1;
+
+            _atr = new AverageTrueRange(atrPeriod);
```
Comment (Martin-Molinero) [inline Indicators/ChandeKrollStop.cs:75]: Let's add the optional `MovingAverageType` for `ChandeKrollStop` to be passed to the ATR 👍

**Review by Martin-Molinero — APPROVED** [review]
Thank you!👍 Last comments requests

File: Indicators/ChandeKrollStop.cs (hunk 8 above)
Comment (femtotrader) [reply Indicators/ChandeKrollStop.cs]: Done

File: Indicators/ChandeKrollStop.cs (hunk 7 above)
Comment (femtotrader) [reply Indicators/ChandeKrollStop.cs]: Done

**Pull request merged**